engines/agi/agi.h

```diff
@@ -491,6 +496,86 @@ struct AgiGame {
 	int16 nonBlockingTextCyclesLeft;
 
 	bool automaticRestoreGame;
+
+	void reset() {
+		_vm = nullptr;
+		adjMouseX = 0;
+		adjMouseY = 0;
+		memset(name, 0, sizeof(name));
+		memset(id, 0, sizeof(id));
+		crc = 0;
+		memset(flags, 0, sizeof(flags));
+		memset(vars, 0, sizeof(vars));
+		horizon = 0;
+		cycleInnerLoopActive = false;
+		cycleInnerLoopType = 0;
+		curLogicNr = 0;
+		execStack.clear();
+		playerControl = false;
+		exitAllLogics = false;
+		pictureShown = false;
+		gameFlags = 0;
+		block.active = false;
+		block.x1 = block.y1 = block.x2 = block.y2 = 0;
+		gfxMode = false;
+		numObjects = 0;
+
+		for (int i = 0; i < ARRAYSIZE(controllerOccured); i++) {
+			controllerOccured[i] = false;
+		}
+
+		for (int i = 0; i < ARRAYSIZE(controllerKeyMapping); i++) {
+			controllerKeyMapping[i].keycode = 0;
+			controllerKeyMapping[i].controllerSlot = false;
+		}
+
+		memset(strings, 0, sizeof(strings));
+
+		for (int i = 0; i < MAX_DIRECTORY_ENTRIES; i++) {
+			memset(&dirLogic[i], 0, sizeof(struct AgiDir));
+			memset(&dirPic[i], 0, sizeof(struct AgiDir));
+			memset(&dirView[i], 0, sizeof(struct AgiDir));
+			memset(&dirSound[i], 0, sizeof(struct AgiDir));
+
+			pictures[i].flen = 0;
```

Comment:
The same (default ctor) is true here. It is imo safer to let the classes define which members must be reset.

Reply:
I'm not sure yet. AgiDir gets cleared outside of the engine constructor in some cases, and I haven't checked if that still only happens on startup. I probably won't have the time to look any further today.